### PR TITLE
feat: add lang code to iframe

### LINKED ISSUE #45050: iFrame titles should be translatable
Now that https://github.com/freeCodeCamp/freeCodeCamp/pull/45014/ is in, we're adding titles of the form 'CatPhotoApp preview' and 'preview'.  This is good from an a11y perspective, but the string needs to be translatable.

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #45050

### CODE REVIEW

**Comment by Sembauke** [comment]
I have looked for the lang code for Spanish (es-419) everywhere. But can't find it.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
We can use the i18n interpolation to pass the project title in.

Might need some tweaking, but here's the general idea:

File: client/src/templates/Challenges/utils/frame.ts [hunk 1]
```diff
@@ -124,7 +129,8 @@ const createFrame =
     const frame = document.createElement('iframe');
     frame.id = id;
     if (typeof title === 'string') {
-      frame.title = title;
+      frame.title = i18next.t('misc.iframe-preview');
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/utils/frame.ts:132]:
```suggestion
      frame.title = i18next.t('misc.iframe-preview', { title });
```

Thank you github for losing my second review comment.

File: client/src/templates/Challenges/utils/frame.ts [hunk 2]
```diff
@@ -124,7 +129,8 @@ const createFrame =
     const frame = document.createElement('iframe');
     frame.id = id;
     if (typeof title === 'string') {
-      frame.title = title;
+      frame.title = i18next.t('misc.iframe-preview', { title });
+      frame.lang = getLangCode(clientLocale);
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/utils/frame.ts:133]: This really does not make sense to me.

The `iframe` only shows English content. So, to tell a browser/screen-reader it should expect anything else is misleading no?

**Review by ShaunSHamilton — DISMISSED** [review]

File: client/src/templates/Challenges/utils/frame.ts (hunk 2 above)
Comment (naomi-lgbt) [reply client/src/templates/Challenges/utils/frame.ts:133]: Yes we should probably set the `lang` attribute to `en` here.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

**Review by naomi-lgbt — DISMISSED** [review]
Heya Sem,

Sorry for the confusion here but I seem to have made this more complicated lol.

We will also need to update https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/src/templates/Challenges/utils/build.ts#L303-L305 to not append ` preview` to the title string - either return the title string or an empty string. This way we can use `preview` in the i18n string instead.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Comment by Sembauke** [comment]
Alright, hopefully things are cleared up now. 

I did not know what I was thinking about the language attribute, sorry!

File: client/src/templates/Challenges/utils/build.ts [hunk 3]
```diff
@@ -300,9 +300,7 @@ export function updateProjectPreview(
     const titleMatch = buildData?.sources?.index?.match(
       /<title>(.*?)<\/title>/
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/utils/build.ts:301]:
```suggestion
      /<title(\s+(\w+=('|"|`)?.*\2)|(\w+))*>(.*?)<\/title\s*>/s
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
I know this has nothing to do with this PR, but I saw the line and could see this biting us in the back.

File: client/src/templates/Challenges/utils/build.ts (hunk 3 above)
Comment (ShaunSHamilton) [reply client/src/templates/Challenges/utils/build.ts:301]: Originally, I was just going to add to allow white space, but then started reading the html spec.

File: client/src/templates/Challenges/utils/build.ts [hunk 4]
```diff
@@ -300,9 +300,7 @@ export function updateProjectPreview(
...
     );
-    const frameTitle = titleMatch
-      ? titleMatch[1].trim() + ' preview'
-      : 'preview';
+    const frameTitle = titleMatch ? titleMatch[1].trim() : '';
```
Comment (ShaunSHamilton) [inline client/src/templates/Challenges/utils/build.ts:303]:
```suggestion
    const frameTitle = titleMatch ? titleMatch[5].trim() : '';
```

**Comment by Sembauke** [comment]
So I am a bit confused, the actual title does not seem to be updated for me. It says `challenge preview`  or `preview preview`.

**Review by ShaunSHamilton — APPROVED** [review]
I am happy with how this behaves

**Review by naomi-lgbt — APPROVED** [review]
Let's get this in.

**Pull request merged**